src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java

```diff
@@ -151,77 +159,131 @@ public void postWithWeirdCharactersIsDisplayedOnDesktopPostDetailsPage() {
+    Assertion.assertFalse(
+        postsCreator.isPostButtonActive(),
+        "User should not be able to add post with only content image uploaded."
+    );
+    postsCreator.removeImage();
 
+    postsCreator.addTitleWith(TextGenerator.defaultText());
+    postsCreator.addDescriptionWith(TextGenerator.defaultText());
+    Assertion.assertFalse(
+        postsCreator.isPostButtonActive(),
+        "User should not be able to add post with only title and description."
+    );
+    postsCreator.clearDescription();
     postsCreator.clearTitle();
-    Assertion.assertFalse(postsCreator.isPostButtonActive(),
-        "User should not be able to add post with only description filled.");
+
+    postsCreator.addTitleWith(TextGenerator.defaultText());
+    postsCreator.addDescriptionWithLink("http://fandom.wikia.com");
+    Assertion.assertFalse(
+        postsCreator.isPostButtonActive(),
+        "User should not be able to add post with only title and open graph."
+    );
     postsCreator.clearDescription();
+    postsCreator.clearOpenGraph();
+    postsCreator.clearTitle();
+
+    postsCreator.addTitleWith(TextGenerator.defaultText());
+    postsCreator.uploadImage();
+    Assertion.assertFalse(
+        postsCreator.isPostButtonActive(),
+        "User should not be able to add post with only title and content image."
+    );
+    postsCreator.removeImage();
+    postsCreator.clearTitle();
 
+    // Add category at the end because there is no way to clear it
     postsCreator.clickAddCategoryButton()
         .findCategoryOn(0)
         .click();
     Assertion.assertFalse(postsCreator.isPostButtonActive(),
-        "User should not be able to add post with only category selected.");
+        "User should not be able to add post with only category.");
 
     postsCreator.addTitleWith(TextGenerator.defaultText());
     Assertion.assertFalse(postsCreator.isPostButtonActive(),
-        "User should not be able to add post with category selected and title filled.");
+        "User should not be able to add post with only category and title.");
+    postsCreator.clearTitle();
+
+    postsCreator.addDescriptionWith(TextGenerator.defaultText());
+    Assertion.assertFalse(
+        postsCreator.isPostButtonActive(),
+        "User should not be able to add post with only category and description."
+    );
+    postsCreator.clearDescription();
+
+    postsCreator.addDescriptionWithLink("http://fandom.wikia.com");
+    Assertion.assertFalse(
+        postsCreator.isPostButtonActive(),
+        "User should not be able to add post with only category and open graph."
+    );
+    postsCreator.clearDescription();
+    postsCreator.clearOpenGraph();
+
+    postsCreator.uploadImage();
+    Assertion.assertFalse(
+        postsCreator.isPostButtonActive(),
+        "User should not be able to add post with only category and content image."
+    );
   }
 
   private CategoryPill fillPostCategoryWith(final PostsCreator postsCreator, final String description) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "fillPostCategoryWith" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)